### DISCUSSION TITLE
无法正常引用参考文献

### DISCUSSION THREAD

**Discussion by libergremlin** [Koyamin/ecnuthesis#15] [Q&A] [answered]
在引用source文件中的thesis-ref.bib中的相关文献时，pdf没法正常输出参考文献，如在正文区输入：\cite{Bayes63:classical},pdf直接在相关句子的右上角显示“Bayes63:classical”，而不是[1]，而且目录中也没有参考文献相关条目出现，请问该怎么解决？

**Comment by Koyamin** [comment, marked as answer]
根据描述来看，像是由于未正确编译导致的。

能否尝试一下编译两次？或者在`<filename>.tex`文件所在的文件夹下使用命令
```
latexmk -xelatex <filename>
```
来编译。

**Reply by libergremlin** [reply]
您好，我还是不能解决这个问题，而且您给的模板生成的pdf也确实没有“参考文献”这一块，请问您能在常见问题模块解决您目录中没有出现参考文献以及论文出现正确引用的这个问题吗？

**Reply by Koyamin** [reply]
`test.pdf`中没有“参考文献”章节是因为`test.tex`中未引用`thesis-ref.bib`文件中的任何参考文献。

经本地测试与 [Overleaf](https://www.overleaf.com/latex/templates/ecnuthesis-latex-thesis-template-for-east-china-normal-university/szppdtkvgvpk) 测试，未能复现你所述的现象。

能否再次检查以下事项：
1. 确保TeX Live保持最新版本；
2. 检查`\ecnuSetup{}`命令的配置项`style/bibResource`正确地引入了所需的`.bib`文件。例如以下的示例：
    ```tex
    \ecnuSetup {
      style = {
        bibResource = {./source/thesis-ref.bib},
      }
    }
    ```
3. 确保正确地使用`\cite{}`来引用`.bib`文件中已存在的参考文献；
4. 确保在最后一章节的内容后使用了`\PrintReference`命令用来打印参考文献:
    ```tex
    % 正文后部分
    \backmatter
    % 导入参考文献 (需要通过 latexmk 编译后才能显示)
    \PrintReference
    ```
5. 确保使用XeLaTeX编译**至少两次**，或使用以下latexmk命令进行编译：
    ```
    latexmk -xelatex <filename>
    ```

另外，你也可以关注编译时是否存在报错信息。若存在报错信息，也可以将这些报错信息附在评论区。

**Reply by libergremlin** [reply]
您是对的，是需要在cmd中在相应文件夹目录下编译：latexmk -xelatex <filename>，texstudio才能正常出现参考文献，请问使用该模板在texstudio中想要正常出现参考文献是必须用该命令进行编译吗？有较为简便的方式吗？

**Reply by libergremlin** [reply]
追问一句，您说在overleaf中没出现我说的情况，可我在overleaf中复现了该情况，由于我对latex的了解甚微，我不知道是哪里出了问题
![捕获](https://github.com/Koyamin/ECNUThesis-Undergraduate/assets/107289146/da819386-6c9e-47b6-ae81-a25ff65b0c0d)

**Reply by Koyamin** [reply]
`test.tex`中使用**相对路径**来告诉LaTeX所需的`.bib`文件的位置，例如示例中的`./source/thesis-ref.bib`就是一个相对路径，开头的`./`意味着命令执行所在文件夹的路径。

在Overleaf中，由于`.bib`文件与`main.tex`处于同一文件夹中，所以`\ecnuSetup{}`命令的配置项`style/bibResource`也要修改为正确的路径。
